File: Observer/go/src/observer/MakingChange.go
package observer

type MakingChange struct {
	son, daughter int
	observers     []Observer
}

func (m *MakingChange) makingSon() {
	m.son++
	m.allNotify()
}

func (m *MakingChange) makingDaughter() {
	m.daughter++
	m.allNotify()
}

func (m *MakingChange) addObserver(observer Observer) {
	m.observers = append(m.observers, observer)
}

func (m *MakingChange) allNotify() {
	for _, observer := range m.observers {
		observer.update(m)
	}
}

File: Observer/python/MakingChange.py

class MakingChange:
    def __init__(self):
        self.son = 0
        self.daughter = 0
        self.observers = []

    def addSon(self):
        self.son+=1
        self.allNotify()
    
    def addDaughter(self):
        self.daughter+=1
        self.allNotify()
    
    def allNotify(self):
        for observer in self.observers:
            observer.update(self)
    
    def addObserver(self, observer):
        self.observers.append(observer)